docs/src/orchid/resources/wiki/components/command.md

```diff
@@ -0,0 +1,167 @@
+and friendly to work with as possible.
+
+You can define commands using the DSL functions available in [extensions](./extension).
+
+#### Defining Commands
+
+Define your commands in your extension's `setup` function. Here's a basic example:
+
+```kotlin
+override suspend fun setup() {
+    command { // this: Command
+        name = "ping"
+
+        action { // this: CommandContext
+            message.channel.createMessage("Pong!")
+        }
+    }
+}
+```
+
+#### Basic Options
+
+All commands require a `name` and an `action`, defined similarly to the above example.
+
+* The `name` of your command should be unique across all installed extensions, and is
+  the name that is used to invoke the command on Discord.
+* The `action` is the body of the command, which will be executed when a user invokes
+  the command.
+
+In addition to these required options, command support a further set of optional
+options:
+
+* `aliases: Array<String>` - This is a list of alternative names that can be used to
+  invoke the command. Note that if a command exists with a `name` that exists as an
+  alias in another command, the command with the matching `name` takes priority.
+  
+  You can have as many aliases as you like.
+* `description: String = """` - This is a description for your command, which will 
+  be displayed by the `help` command. Try to make it descriptive, while also keeping 
+  it short!
+* `enabled: Boolean = true` - If you disable a command by setting this to `false`, it 
+  cannot be invoked and will not be shown in the `help` command. This can be modified 
+  at runtime, if required.
+* `hidden: Boolean = false` - If you want it to be possible to invoke your command, 
+  but would still prefer it to be hidden from the `help` command, then you can set this
+  to `true`.
+* `signature: String = ""` - The command's signature represents how the arguments to the
+  command should be formatted. For the sake of consistency, all optional arguments should
+  be specified like this: `[name]`. Required arguments should use `<name>`. Lists can
+  be either required or optional, but should contain an ellipsis - `<name ...>`.
+  
+  The signature will be shown in help commands, and may also be automatically generated from a
+  data class. See the below section on command arguments for more information on signature
+  generation.
+
+#### Checks
+
+You will likely want to write commands that are only invoked based on a set of conditions.
+In order to facilitate this in a manner that allows you to easily reuse these conditions,
+commands support a [checks system](check).
+
+```kotlin
+suspend fun defaultCheck(event: MessageCreateEvent): Boolean {
+    with(event) {
+        return when {
+            message.author?.id == bot.kord.getSelf().id -> false  // Check that we didn't send this message.
+            message.author!!.isBot == true              -> false  // Check that another bot didn't send this message.
```

Comment:
Why a safe call followed by a not-null assertion? You seem to be using both through the wiki pages, it would be nice to select one.

Reply:
It was still appropriate to use `!!` in some cases (or just for simplification), but I've updated most instances.